cli_tools/google-osconfig-agent/config/config.go

```diff
@@ -37,19 +51,119 @@ var (
 	oauth            = flag.String("oauth", "", "path to oauth json file")
 	debug            = flag.Bool("debug", false, "set debug log verbosity")
 
-	googetRepoFilePath = flag.String("googet_repo_file", "C:/ProgramData/GooGet/repos/google_osconfig.repo", "googet repo file location")
-	zypperRepoFilePath = flag.String("zypper_repo_file", "/etc/zypp/repos.d/google_osconfig.repo", "zypper repo file location")
-	yumRepoFilePath    = flag.String("yum_repo_file", "/etc/yum/repos.d/google_osconfig.repo", "yum repo file location")
-	aptRepoFilePath    = flag.String("apt_repo_file", "/etc/apt/sources.list.d/google_osconfig.list", "apt repo file location")
+	agentConfig   *config
+	agentConfigMx sync.RWMutex
+	version       string
+)
 
-	osPackageEnabled bool
-	osPatchEnabled   bool
+type config struct {
+	osInventoryEnabled, osPackageEnabled, osPatchEnabled                     bool
+	googetRepoFilePath, zypperRepoFilePath, yumRepoFilePath, aptRepoFilePath string
+}
 
-	version string
-)
+func getAgentConfig() config {
+	agentConfigMx.RLock()
+	defer agentConfigMx.RUnlock()
+	return *agentConfig
+}
+
+func enabled(s string) bool {
+	enabled, err := strconv.ParseBool(s)
+	if err != nil {
+		// Bad entry returns as not enabled.
+		return false
+	}
+	return enabled
+}
+
+type metadataJSON struct {
+	Instance instanceJSON
+	Project  projectJSON
+}
+
+type instanceJSON struct {
+	Attributes attributesJSON
+}
+
+type projectJSON struct {
+	Attributes attributesJSON
+}
+
+type attributesJSON struct {
+	OSInventoryEnabled string `json:"os-inventory-enabled"`
+	OSPatchEnabled     string `json:"os-patch-enabled"`
+	OSPackageEnabled   string `json:"os-package-enabled"`
+}
+
+func (c *config) setFromMetadata(md metadataJSON) {
```

Comment:
minor: it might be clearer to change this to `buildFromMetadata()` or `createFromMetadata()` and have it return the new object. The advantage being that it would no longer mutate the state of an object that may also be managed in other places.

That would also allow you to set the defaults in the same place where you check for overrides which would make the logic a lot clearer.